cmd/main.go

```diff
@@ -34,45 +35,33 @@ func main() {
 	if err := initConfig(); err != nil {
 		log.Fatalf("error initializing configs: %s", err.Error())
 	}
-	dbConfig := repository.DBConfig{
-		Host:       viper.GetString("db.host"),
-		Port:       viper.GetString("db.port"),
-		Username:   viper.GetString("db.username"),
-		Password:   viper.GetString("db.password"),
-		DBName:     viper.GetString("db.name"),
-		SSLMode:    viper.GetString("db.ssl_mode"),
-		DriverName: viper.GetString("db.driver_name"),
-	}
-	mailConfig := service.MailConfig{
-		Host:     viper.GetString("mail.host"),
-		Port:     viper.GetString("mail.port"),
-		Username: viper.GetString("mail.username"),
-		Password: viper.GetString("mail.password"),
+	conf, err := configs.NewConfig("configs/config.yml")
+	if err != nil {
+		log.Fatalf("error initializing configs: %s", err.Error())
 	}
-	db, err := repository.NewDB(dbConfig)
+	log.Info(conf.Mail.Host)
+	db, err := repository.NewDB(conf.DB)
 	if err != nil {
 		log.Fatalf("failed to initialize db: %s", err.Error())
 	}
 	driver, err := postgres.WithInstance(db.DB, &postgres.Config{})
 	if err != nil {
 		log.Fatalf("failed to create postgres driver: %s", err.Error())
 	}
-	m, err := migrate.NewWithDatabaseInstance(
-		MigrationsPath,
-		dbConfig.DBName, driver)
+	m, err := migrate.NewWithDatabaseInstance(MigrationsPath, conf.DB.DBName, driver)
 	if err != nil {
 		log.Fatalf("failed to create migration instance: %s", err.Error())
 	}
 	err = m.Up()
 	if err != nil {
 		if errors.Is(err, migrate.ErrNoChange) {
-			log.Print("No migrations to apply")
+			log.Info("No migrations to apply")
 		} else {
 			log.Fatalf("failed to apply migrations: %s", err.Error())
 		}
 	}
 	repositories := repository.NewRepository(db)
-	services := service.NewService(repositories, mailConfig)
+	services := service.NewService(repositories, conf)
```

Comment:
I don't see a reason to aggregate all services into one struct. Why not to just instantiate concerete services in main.go and pass specific service as a parameter?

Reply:
I guess it simplifies dependency injection. I can only pass one struct that contains all services instead of passing multiple services individually. And in future if I need to add new service, only service struct would be changed.
But I'm going to rewrite my code using uber dig library, so it'll be cleaner

Reply:
But if you need to add a new service you need to use it somewhere in the project. And here is the drawback - any part of your app can use any service form the service struct.
Usually on practise there are many services in the app. And with such approach it will be hard to find required service. Besides it leads to tighter coupling in the system cause each part of the app that has service struct could call any of available services. It's hard to control where and what service is used. And if it's used in a correct way

Reply:
Agree, thx, I'll fix this later